inference-engine/src/gna_plugin/gna_graph_compiler.cpp

```diff
@@ -1193,8 +1193,8 @@ void GNAGraphCompiler::EltwisePrimitive(InferenceEngine::CNNLayerPtr layer) {
         THROW_GNA_LAYER_EXCEPTION(layer) << " Inputs size mismatch " << in_4b_total_size << " != " << in_2b_total_size;
     }
 
-    uint32_t num_rows_in = in_4b_channels * in_4b_height * in_4b_width;
-    uint32_t num_columns_in = in_4b_batch;
+    uint32_t num_rows_in = in_4b_total_size;
+    uint32_t num_columns_in = 1;
```

Comment:
These variables are needed to check if batch sizes are equal. They may be not equal to 1 but equal one with another, in this case data can be reshaped to have columns = 1 and rows = total data size. Comment is added to explain it. 
Actually the previous behavior was the same but not so explicit since getFromIRDimsOrderNCHW and FROM_IR_DIM always returned batch size 1 for 2D data that doesn't seem to be always correct.

Reply:
OK